### PR TITLE
GOV-301: Mock async Authorization API

### PR DESCRIPTION
**Pull request by arkadasfynarfin**
Jira link: [GOV-301](https://mifosforge.jira.com/browse/GOV-301)

Create new authorization API which responds with mock response asynchronously.

**NOTE**:  Response is dependant on amount sent in the request body. If amount is greater than threshold amount, response is failed otherwise, it is successful.

Expected response: `202`

[GOV-301]: https://mifosforge.jira.com/browse/GOV-301?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by arkadasfynarfin** [comment]
@fynmanoj Please review this PR. Upstream is now set to Open MF.

File: src/main/java/org/mifos/connector/mockPaymentSchema/api/BatchApi.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+public class BatchApi {
+
+    @Autowired
+    private BatchService batchService;
+
+    @PostMapping("/batches/{batchId}")
```
Comment (fynmanoj) [inline src/main/java/org/mifos/connector/mockPaymentSchema/api/BatchApi.java:16]: this doesn't follow API guidelines.
A POST API usually doesn't have id in path

File: src/main/java/org/mifos/connector/mockPaymentSchema/service/BatchService.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            response.setClientCorrelationId(clientCorrelationId);
+            response.setReason("Error getting authorization for the request");
+        } else {
+            response.setClientCorrelationId(clientCorrelationId);
+            response.setStatus("Y");
+        }
+        try {
+            sendCallbackService.sendCallback(new ObjectMapper().writeValueAsString(response), callbackUrl);
+        } catch (JsonProcessingException e) {
+            e.printStackTrace();
```
Comment (fynmanoj) [inline src/main/java/org/mifos/connector/mockPaymentSchema/service/BatchService.java:45]: donot print stacktrace , use logger instead

**Review by fynmanoj — CHANGES_REQUESTED** [review]

File: src/main/java/org/mifos/connector/mockPaymentSchema/api/BatchApi.java (hunk 1 above)
Comment (arkadasfynarfin) [reply src/main/java/org/mifos/connector/mockPaymentSchema/api/BatchApi.java:16]: @fynmanoj this is resolved. Would be adding check for command = authorize in the method.

File: src/main/java/org/mifos/connector/mockPaymentSchema/api/BatchApi.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    public ResponseEntity<Object> getAuthorization(@PathVariable String batchId,
+            @RequestHeader("X-Client-Correlation-ID") String clientCorrelationId, @RequestBody AuthorizationRequest authorizationRequest,
+            @RequestParam(value = "command", required = false, defaultValue = "authorize") String command,
+            @RequestHeader(value = "X-CallbackURL") String callbackURL) {
+        try {
```
Comment (arkadasfynarfin) [inline src/main/java/org/mifos/connector/mockPaymentSchema/api/BatchApi.java:21]: add check for command value. If not authorize return not found.

File: src/main/java/org/mifos/connector/mockPaymentSchema/api/BatchApi.java (hunk 1 above)
Comment (fynmanoj) [reply src/main/java/org/mifos/connector/mockPaymentSchema/api/BatchApi.java:16]: command is used

File: src/main/java/org/mifos/connector/mockPaymentSchema/api/BatchApi.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+            batchService.getAuthorization(batchId, clientCorrelationId, authorizationRequest, callbackURL);
```
Comment (fynmanoj) [inline src/main/java/org/mifos/connector/mockPaymentSchema/api/BatchApi.java:22]: add check for the command and donot accept any other command. Donot set default value for the command

**Review by fynmanoj — CHANGES_REQUESTED** [review]

**Review by fynmanoj — APPROVED** [review]

**Pull request merged**